inference/services/KernelFactory.cs

```diff
@@ -39,35 +50,40 @@ private Kernel CreateKernel(HttpClient httpClient)
         return kernalBuilder.Build();
     }
 
-    public async Task<Kernel> GetOrCreateKernelForInferenceAsync(CancellationToken cancellationToken = default)
+    public async Task<Kernel> GetOrCreateKernelForInferenceAsync(CancellationToken cancellationToken = default, int? index = null)
     {
-        if (this.kernelForInference is not null) return this.kernelForInference;
+        this.kernelsForInference ??= new List<Kernel>(config.CHAT_LLM_CONNECTION_STRINGS.Length);
```

Comment:
You cannot have this outside of the semaphore because you could end up having a race condition.

Reply:
updated.